lib/modules/datasource/npm/get.ts

```diff
@@ -171,26 +171,22 @@ export async function getDependency(
       return release;
     });
     logger.trace({ dep }, 'dep');
-    // serialize first before saving
-    // TODO: use dynamic detection of public repos instead of a static list (#9587)
-    const whitelistedPublicScopes = [
-      '@graphql-codegen',
-      '@storybook',
-      '@types',
-      '@typescript-eslint',
-    ];
     if (
-      !raw.authorization &&
-      (whitelistedPublicScopes.includes(packageName.split('/')[0]) ||
-        !packageName.startsWith('@'))
+      (raw.headers?.['cache-control'] || '')
```

Comment:
```suggestion
      (raw.headers?.['cache-control'] ?? '')
```

would a regex be faster? 🤔 

```js
/(^|,)\s*public\s*(,|$)/
```